### DISCUSSION TITLE
Optional Trait evaluation in a Segment

### DISCUSSION THREAD

**Discussion by dariomncs** [Flagsmith/flagsmith#1281] [Q&A] [answered]
Hello!
I was trying to implement a Segment with a "% Split" excluding users with a specific trait, and as it is a new trait, few users have it.

What I found is that apparently the condition "_doesn't match !=_" implies a "_trait exists_", as all the users are appearing excluded when i add the AND condition, and re-included (following the other condition) when I remove it.

I'm doing something bad? There is a way to include users that doesn't have a specific trait, consider a trait optional or setup a trait value by default? 
 
![image](https://user-images.githubusercontent.com/495006/177745526-e8dec8bb-5dcf-44f9-8633-a6a4606e6d02.png)

**Comment by dabeeeenster** [comment, marked as answer]
Hi @dario1985 - we don't currently have any rules that operate on the existence of a trait value. So what you are seeing is intended behaviour, but I can appreciate it doesn't work for your use case.

We do have this outstanding https://github.com/Flagsmith/flagsmith/issues/1145 which we will try and get to!
